### PR TITLE
refs #23211 - foreman-proxy: puppetca token whitelisting

### PR DESCRIPTION
**Pull request by timogoebel**
Packaging PR for https://github.com/theforeman/smart-proxy/pull/592. This is only relevant for nightly.

### CODE REVIEW

**Comment by timogoebel** [comment]
@ekohl: rpm tests are failing. Do you have an idea, why? I can't make much sense of it.

**Comment by ekohl** [comment]
This has something to do with our pipeline. We run repoclosure on all packages with a single directory for all downloaded RPMs. Then it has the lookaside repositories just for non-scl when checking foreman-proxy. That means it can't find the SCL ones for tfm-rubygem-jwt.

@ehelms thoughts?

The current workaround would be to spit it into 2 PRs: one to move JWT to scl_and_nonscl and the other to modify foreman-proxy.

**Comment by timogoebel** [comment]
> The current workaround would be to spit it into 2 PRs: one to move JWT to scl_and_nonscl and the other to modify foreman-proxy.

I can do that, no worries. Please just let me know if I should start.
Thanks for the explanation.

**Comment by timogoebel** [comment]
> one to move JWT to scl_and_nonscl

--> #2999 

> the other to modify foreman-proxy

I will rebase this PR after ^^ is merged.

**Comment by ekohl** [comment]
At this point I think it failed because the RPM isn't on any mirrors yet. I'll have a look on Monday.

**Comment by timogoebel** [comment]
[test rpm]

**Comment by timogoebel** [comment]
@ekohl: Looks like this is still failing, but I have no clue why. Do you mind? 👼

**Comment by mmoll** [comment]
I think there's no non-SCL-build of rubygem-jwt available in the repo, was there an appropriate PR?

**Comment by timogoebel** [comment]
> I think there's no non-SCL-build of rubygem-jwt available in the repo, was there an appropriate PR?

@mmoll: #2999 should have taken care of this.

**Comment by mmoll** [comment]
I see, I guess we need to wait for https://projects.theforeman.org/issues/24955 then.

**Comment by timogoebel** [comment]
[test rpm]

**Comment by ekohl** [comment]
https://github.com/theforeman/foreman-packaging/pull/3010 will unstick this

**Comment by ehelms** [comment]
[test rpm]

**Comment by ekohl** [comment]
The package wasn't released because it wasn't added in koji. Done now. Probably needs a mashing of nightly before repoclosure works now.

**Comment by timogoebel** [comment]
[test rpm]

**Comment by timogoebel** [comment]
Tests are finally green.

File: packages/foreman/foreman-proxy/foreman-proxy.spec [hunk 1]
```diff
@@ -132,6 +137,8 @@ ln -sv %{_tmppath} %{buildroot}%{_datadir}/%{name}/tmp
 %doc README.md VERSION
 %license LICENSE
 %{_datadir}/%{name}
+%dir %{_sharedstatedir}/%{name}
```
[the remark this answers is not in the document]
Comment (timogoebel) [reply packages/foreman/foreman-proxy/foreman-proxy.spec]: It does, let me remove this.

**Pull request merged**